olga/charts/tests/test_views.py

```diff
@@ -0,0 +1,269 @@
+        tabular_format_countries_list = [['Unset', 0, 0]]
+
+        result = MapView.get_statistics_top_country(tabular_format_countries_list)
+
+        self.assertEqual(None, result)
+
+
+@patch('olga.analytics.models.InstallationStatistics.objects.aggregate')
+class TestViewsHelpFunctions(TestCase):
+    """
+    Tests for charts help functions.
+    """
+
+    def test_update_datetime_if_data_exists(self, mock_installation_statistics_model_objects_aggregate):
+        """
+        Verify that get_first_and_last_datetime_of_update_data method returns first and last objects datetime.
+        """
+        mock_min_datetime_of_update_data = datetime(2017, 6, 1, 14, 56, 18)
+        mock_max_datetime_of_update_data = datetime(2017, 7, 2, 23, 12, 8)
+
+        mock_installation_statistics_model_objects_aggregate.return_value = {
+            'data_created_datetime__min': mock_min_datetime_of_update_data,
+            'data_created_datetime__max': mock_max_datetime_of_update_data
+        }
+
+        result = get_data_created_datetime_scope()
+
+        self.assertEqual(
+            (mock_min_datetime_of_update_data, mock_max_datetime_of_update_data), result
+        )
+
+    @patch('olga.charts.views.datetime')
+    def test_update_datetime_if_no_data(
+            self, mock_datetime, mock_installation_statistics_model_objects_aggregate
+    ):
+        """
+        Verify that get_first_and_last_datetime_of_update_data method returns `datetime.now` if no objects.
+        """
+        mock_first_datetime_of_update_data = mock_last_datetime_of_update_data = datetime(2017, 7, 2, 23, 12, 8)
+
+        mock_datetime.now.return_value = mock_first_datetime_of_update_data
+
+        mock_installation_statistics_model_objects_aggregate.return_value = {
+            'data_created_datetime__min': None,
+            'data_created_datetime__max': None
+        }
+
+        result = get_data_created_datetime_scope()
+
+        self.assertEqual(
+            (mock_first_datetime_of_update_data, mock_last_datetime_of_update_data), result
+        )
+
+
+class TestGraphsView(TestCase):
+    """
+    Tests for map view, that contains graphs statistics functionality.
+    """
+
+    def setUp(self):
+        """
+        Set up common response.
+        """
+        self.response = self.client.get('/')
+
+    def test_graphs_url(self):
+        """
+        Verify that graphs url resolves to graphs view.
+        """
+        self.assertEqual(self.response.resolver_match.func.__name__, GraphsView.__name__)
+
+    def test_graphs_view_status_code(self):
+        """
+        Verify that graphs view returns status 200 (ok) after get request to graphs url.
+        """
+        self.assertEqual(self.response.status_code, 200)
+
+    def test_graphs_view_html(self):
+        """
+        Verify that graphs view uses correct html-template after get request to graphs url.
+        """
+        self.assertTemplateUsed(self.response, 'charts/graphs.html')
```

Comment:
I answer for all comments this type: [this](https://github.com/raccoongang/acceptor/pull/40/files#r128314700), [this](https://github.com/raccoongang/acceptor/pull/40/files#r128315050), [this](https://github.com/raccoongang/acceptor/pull/40/files#r128315530) and [this](https://github.com/raccoongang/acceptor/pull/40/files#r128315575).

I think 20 code lines costs dev-team's confidence. Anyone know that no one of developer in teams did not change something in repository without local testing (if it happens `CI` will raise issue).